### DISCUSSION TITLE
Login process

### DISCUSSION THREAD

**Discussion by cyclotron3k** [cardillan/mindcode#26] [General]
Just some thoughts on the new login/save experience:

- register should have two password boxes (one for password confirmation)
- login experience should be accessible from register experience and vise-versa
- seeing an unhandled exception when registering with a previously used email address
- no "wrong password" notification displayed when password entered incorrectly
- seem to get a stack trace every time I hit compile

For login, have you thought about using an SSO/Authentication provider like Auth0?

**Comment by francois** [comment]
Hey, thanks for taking the time to look at this. I did think of using Google OAuth or something similar, but then I decided against it. If Google closes down your account, you lose access to Mindcode, in addition to everything else you used Google for. This is the same problem with all other OAuth providers.

Of course, Mindcode is not really important in the grand scheme of things, but that was added complexity which I didn't want to support. Plus, if I add Google, I'll probably add Facebook, and then Apple, and then... this is a never-ending cycle. If I had chosen Auth0, people who wanted to try Mindcode would have had to register with Auth0, an unnecessary friction.

As to all your other points, they are very valid, and are simply not handled at the moment. The user-experience (UX) will be improved. I wanted to get this in front of people, to see if it was something they wanted / needed.

Thanks again for your time. I'll work on those in due time.

**Reply by cyclotron3k** [reply]
No problem, glad I could help.

Just on Auth0; you don't actually have to register to log in with Auth0, you just select which SSO providers you want to use (Google, Facebook, LinkedIn, Github, etc, etc,) and Auth0 takes care of setting up and maintaining all the integrations for you.

Anyway, I understand your decision. No problem :+1:
